app/controllers/dynamic_pages_controller.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+class DynamicPagesController < ApplicationController
+  before_action :require_admin, except: [:index, :show]
+  def index
+    if logged_in?
+      @current_user_is_admin = is_admin?
+      @current_user_is_teacher = is_teacher?
+      @current_user_is_verified_teacher = is_verified_teacher?
+    end
+    @all_dynamic_pages = DynamicPage.all
+  end
+  def delete
+    DynamicPage.destroy(params[:id])
+    redirect_to dynamic_pages_path
+  end
+  def new
+    @dynamic_page = DynamicPage.new(flash[:dynamic_page])
+  end
+  def create
+    @dynamic_page = DynamicPage.new(dynamic_page_params)
+    if DynamicPage.find_by(slug: @dynamic_page.slug)
+      flash[:dynamic_page] = params[:dynamic_page]
+      redirect_to({ action: "new" }, alert:  "That slug already exists :(")
+    elsif @dynamic_page.save
+      flash[:success] = "Created #{@dynamic_page.title} page successfully."
+      redirect_to ({ action: "show", slug: @dynamic_page.slug })
+    else
+      redirect_to root_path, alert: "Failed to submit information :("
+    end
+  end
+  def show
+    @dynamic_page = DynamicPage.find_by(slug: params[:slug])
+    if @dynamic_page.permissions == "Admin" && !is_admin?
```

Comment:
Instead of comparing these to a string value we should have a method on a `dynamic_page`.

Perhaps:

```
class DynamicPage
  def admin_only?
       permissions == 'Admin'
  end
end
```